tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorTests.cs

```diff
@@ -120,6 +120,118 @@ public void Any_with_a_predicate_on_documents()
                 "{ G : { $elemMatch : { D : \"Don't\", 'E.F' : 33 } } }");
         }
 
+        [Fact]
+        public void Any_with_a_predicate_on_document_itself()
+        {
+            Assert(
+                x => x.G.Any(g => g != null),
+                2,
+                "{ 'G' : { '$elemMatch' : { '$ne' : null } } }");
+
+            Assert(
+                x => x.G.Any(g => null != g),
+                2,
+                "{ 'G' : { '$elemMatch' : { '$ne' : null } } }");
+
+            Assert(
+                x => x.G.Any(g => g.E.I.Any(i => i == "insecure")),
+                1,
+                "{ \"G.E.I\" : { '$elemMatch' : { '$eq': 'insecure' } } }");
+
+            // todo: the generated query is wrong: {G.E.I= { "$elemMatch" : { "$or" : [{ "" : "insecure" }, { "" : "igloo" }] } }}.
```

Comment:
@jyemin, @rstam. Should I create a ticket about these situations?

Reply:
In general, TODO's should be ticketed.  When they are just in the code they have a tendency to never get done.

Reply:
Yep, I am going to create a ticket/tickets before I close this one. I just wanted to confirm that it's a good idea.

Reply:
I don't like the idea of having a lot of comments headed with "generated query is wrong".

Can you just put these in a JIRA ticket instead of as comments here?

Reply:
yep, It's my plan. This todo tests just to show the problem before I created a ticket